backend/clubs/views.py

```diff
@@ -2878,33 +2878,64 @@ def get(self, request):
                                     type: integer
         ---
         """
-        id = request.query_params.get("event")
-        if not id:
-            return Response({"count": 0})
-
-        try:
-            conn = get_redis_connection()
-        except NotImplementedError:
-            return Response({"count": 0})
-        key = f"zoom:meeting:live:{id}"
-        ans = conn.get(key)
-        if ans is None:
-            ans = 0
-        else:
-            ans = int(ans)
-        return Response({"count": ans})
+        meeting_id = request.query_params.get("event")
+        return Response({"count": ZoomMeetingVisit.objects.filter(meeting_id=meeting_id).count()})
 
     def post(self, request):
         action = request.data.get("event")
-        meeting_id = request.data.get("payload", {}).get("object", {}).get("id", None)
-        if meeting_id is not None:
-            conn = get_redis_connection()
-            key = f"zoom:meeting:live:{meeting_id}"
-            if action == "meeting.participant_joined":
-                conn.incr(key)
-            elif action == "meeting.participant_left":
-                conn.decr(key)
-            conn.expire(key, datetime.timedelta(hours=8))
+        if action == "meeting.participant_joined":
+            email = (
+                request.data.get("payload", {})
+                .get("object", {})
+                .get("participant", {})
+                .get("email", None)
+            )
+            person = get_user_model().objects.filter(email=email)
+
+            event_name = request.data.get("payload", {}).get("object", {}).get("topic", None)
+            event = Event.objects.filter(name=event_name).order_by("-start_time")[0]
```

Comment:
My logic here is that we can get the event by name and then the most recent one will be the correct event (because the participant is currently in the event) but this might not work with recurring events that are planned into the future (because order by DESC means those will come first). There has got to be a better way here so lmk if anyone has ideas

Reply:
This may be too fragile, the topic isn't guaranteed to be the event name and the topic can always be changed by the user.

You should instead match on Zoom meeting IDs, which can't be changed as easily. See if you can match based on the ID in the Zoom URL.